homeassistant/components/sensor/date_reminder.py

```diff
@@ -0,0 +1,95 @@
+"""
+This component provides HA sensor for date reminder
+
+For more details about this sensor, please refer to the documentation at
+https://home-assistant.io/components/sensor.date_reminder/
+"""
+
+import datetime
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_DAYS = 'days_since'
+ATTR_REMAINING = 'days_remaining'
+
+DEFAULT_NAME = 'Date Reminder'
+CONF_START_DATE = 'start_date'
+CONF_REMINDER = 'days_in_future'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=60)
```

Comment:
Doesn't make sense to run every hour as your are counting days. An improvement could be to only run once per day (right after midnight) and when you start Home Assistant.